src/basic/cgroup-util.c

```diff
@@ -2716,11 +2717,20 @@ bool cg_is_unified_wanted(void) {
         if (cg_unified_flush() >= 0)
                 return (wanted = unified_cache >= CGROUP_UNIFIED_ALL);
 
-        /* Otherwise, let's see what the kernel command line has to say.
-         * Since checking is expensive, cache a non-error result. */
+        /* If we passed cgroup_no_v1=all, it seems highly unlikely that we
+         * want to use hybrid or legacy hierarchy. */
+        wanted = is_default;
+        r = proc_cmdline_get_key("cgroup_no_v1", 0, &c);
+        if (r > 0 && streq_ptr(c, "all"))
+                wanted = true;
+
+        /* ...but, let's see what the kernel command line has to say. Maybe
+         * we were explicitly passed systemd.unified_cgroup_hierarchy. */
         r = proc_cmdline_get_bool("systemd.unified_cgroup_hierarchy", &b);
+        if (r > 0)
+                wanted = b;
```

Comment:
This should be `r >= 0`, no?

Reply:
I don't think so? `proc_cmdline_get_bool` returns 0 on "not found", as passed through from `proc_cmdline_get_key`

Reply:
Right.